zeebe/engine/src/test/java/io/camunda/zeebe/engine/state/deployment/FormStateTest.java

```diff
@@ -66,4 +75,52 @@ void shouldReturnEmptyIfNoFormIsDeployedForFormIdAndVersionTag() {
     // then
     assertThat(persistedForm).isEmpty();
   }
+
+  @Test
+  void shouldIterateThroughAllForms() {
+    // give -- three forms exist in the state
+    final var form1 = createFormRecord(1);
+    final var form2 = createFormRecord(2);
+    final var form3 = createFormRecord(3);
+
+    formState.storeFormInFormColumnFamily(form1);
+    formState.storeFormInFormColumnFamily(form2);
+    formState.storeFormInFormColumnFamily(form3);
+
+    // when -- iterating through all forms
+    final var visitor = Mockito.mock(PersistedFormVisitor.class);
+    when(visitor.visit(any())).thenReturn(true);
+    formState.forEachForm(null, visitor);
+
+    // then -- visited all three forms
+    verify(visitor, times(3)).visit(any());
```

Comment:
🔧 Same comments as in the processes PR. It'd be cleaner in these tests to actually verify the arguments using a captor to make sure you're not skipping the wrong one.